dvc/external_repo.py

```diff
@@ -123,21 +123,32 @@ def _set_cache_dir(self):
 
         self.cache.local.cache_dir = cache_dir
 
-    def _set_upstream(self):
-        # check if the URL is local and no default remote is present
-        # add default remote pointing to the original repo's cache location
+    def _fix_upstream(self):
         if os.path.isdir(self.url):
-            if not self.config["core"].get("remote"):
-                src_repo = Repo(self.url)
-                try:
-                    cache_dir = src_repo.cache.local.cache_dir
-                finally:
-                    src_repo.close()
-
-                self.config["remote"]["auto-generated-upstream"] = {
-                    "url": cache_dir
-                }
-                self.config["core"]["remote"] = "auto-generated-upstream"
+            upstream_name = self.config["core"].get("remote")
+            src_repo = Repo(self.url)
+            try:
+                if upstream_name:
+                    self._fix_local_remote(src_repo, upstream_name)
+                else:
+                    self._add_upstream(src_repo)
+            finally:
+                src_repo.close()
+
+    def _fix_local_remote(self, src_repo, remote_name):
+        # Keep the relative path for a local remote relative
```

Comment:
I think the comment is not relevant:
1. Cause the paths are absolute as `Config.RelPath`
2. There are no any conversions of the paths

Reply:
Ok, I guess the wording is not correct in that comment. The path is absolute, but it was built from a relative URL, and thats why it is being changed with the path from the source repo.

What do you think of making the test a check to see if the path is relative instead?

```python
if isinstance(new_remote["url"], RelPath):
    ...
```

This makes it explicit that what we care about there is paths with urls relative to the repository, with no more clarification required.

Reply:
@tizoc maybe `if not os.path.isabs(new_url)` then? It would work with strings and pathlib objects. Just worried about this being easilly broken with some hand-written dict config in tests or elsewhere.

Reply:
Ok, I had not thought about that. But you are right, `if not os.path.isabs(new_url)` would work the same and without those possible breakages. The benefit of checking for `RelPath` instances was only to better express intent.